src/to_ascii.cpp

```diff
@@ -44,13 +55,13 @@ constexpr static uint8_t is_forbidden_domain_code_point_table[] = {
 
 static_assert(sizeof(is_forbidden_domain_code_point_table) == 256);
 
-inline bool is_forbidden_domain_code_point(const char c) noexcept {
-  return is_forbidden_domain_code_point_table[uint8_t(c)];
+constexpr bool is_forbidden_domain_code_point(const char c) noexcept {
+  return is_forbidden_domain_code_point_table[static_cast<uint8_t>(c)];
```

Comment:
I really don't mind that `static_cast<uint8_t>(c)` thing, and I know that it is what Node seems to require... But practically speaking, there is no difference between `static_cast<uint8_t>(c)`  and `uint8_t(c)` except that one is uglier. It is not faster or safer to do  `static_cast<uint8_t>(c)`. I am not sure why people do these things.

Reply:
Yes, I don't have any strong opinions here. Clang-tidy gives warnings since previous usage was C syntax, and we are writing C++. I can remove it :+1:

Reply:
@anonrig I am specifically *not* asking your to change this. I just want to share my opinion and maybe, over time, influence people to rethink these ideas.

Note that `uint8_t(c)` is C++. It will not build in C. In C, you need to write `(uint8_t)c` which is ugly in my opinion.

They introduced the verbose static cast approach to make it very visible that there is casting as it is, in general, unsafe. 

So... `(MyCustomClass)c` is kind of bad, especially when `MyCustomClass` has derived classes.. So you should use `static_cast<MyCustomClass>(c)` to make it impossible to miss what you are doing. But for number types, it is a different story. Everyone understands what `uint8_t(c)` does.